### PR TITLE
engine.ex docs: intercalating -> interlacing

### PR DESCRIPTION
**Pull request by henrik**
I know a lot of words but I had to look up "intercalating" :D Maybe this is clearer?

### CODE REVIEW

**Comment by josevalim** [comment]
I will let @chrismccord decide. :D

**Comment by henrik** [comment]
Maybe a woodworking term, Chris? "Dovetailing"? :D

**Comment by seb3s** [comment]
It has not the same meaning. Intercalating is closest to what's happening here (imho)

**Comment by mcrumm** [comment]
TIL ...and it does seem a fitting definition of what's happening here.

[**intercalate**](https://www.merriam-webster.com/dictionary/intercalate) _verb_
> to insert or position between or among existing elements or layers 

- [Intercalation (biochemistry)](https://en.wikipedia.org/wiki/Intercalation_(biochemistry)) - the insertion of molecules between the planar bases of deoxyribonucleic acid (DNA).
- [Intercalation (chemistry)](https://en.wikipedia.org/wiki/Intercalation_(chemistry)) - the reversible inclusion or insertion of a molecule (or ion) into materials with layered structures.

**Comment by josevalim** [comment]
Oh wow, I had no idea it is uncommon. "intercalar" in portuguese is somewhat normal. :)

**Comment by jrissler** [comment]
I've never heard of it either - but expanding people's vocabulary as part of the documentation is never a bad thing IMO!

**Comment by seb3s** [comment]
The same in french, very common  🙂

**Comment by henrik** [comment]
My impression from looking it up is that you can intercalate a thing between others but if seemed less clear that you can intercalate two things together. And either way it’s a word I’ve never heard as a lifelong word geek, so I’m thinking it may hurt readability even if technically accurate. 

But I recognise the risk of a long bikeshedding session here so I will try my best to leave it there :D Not an important change and I could happily live with either.

**Comment by josevalim** [comment]
At least in portuguese, intercalate is precise in that intercalating AAAA with BBBB means ABABABAB. A zipper would also be a proper analogy but unfortunately zipping means building pairs in functional languages.

**Comment by seb3s** [comment]
I guess it then could be replaced by the more common "alternating" word. Still exact and probably known by every coder.

**Comment by josevalim** [comment]
:+1: on alternating.

File: lib/phoenix_live_view/engine.ex [hunk 1]
```diff
@@ -173,7 +173,7 @@ defmodule Phoenix.LiveView.Engine do
     5. a `Phoenix.LiveView.Component` struct, see "Component" below
 
   When you render a `.leex` template, you can convert the
-  rendered structure to iodata by intercalating the static
+  rendered structure to iodata by interlacing the static
```
Comment (josevalim) [inline lib/phoenix_live_view/engine.ex:176]:
```suggestion
  rendered structure to iodata by alternating the static
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by henrik** [comment]
❤️💚❤️💚❤️

**Pull request merged**